data/tpch_loader.py

```diff
@@ -0,0 +1,467 @@
+                batch_size=1,
+                runtime=EventTime(runtime, EventTime.Unit.US),
+            ),
+        )
+        return WorkProfile(
+            name=f"{query_name}_{node_name}_execution_profile",
+            execution_strategies=execution_strategies,
+        )
+
+    def get_next_workload(self, current_time: EventTime) -> Optional[Workload]:
+        # Reset rng if this is the first workload. This is to ensure we have
+        # parity with how jobs are spawned in Spark
+        if self._current_release_pointer == 0:
+            self._rng = random.Random(self._rng_seed)
+
+        to_release = []
+        while (
+            self._current_release_pointer < len(self._query_nums_and_release_times)
+            and self._query_nums_and_release_times[self._current_release_pointer][1]
+            <= current_time + self._workload_update_interval
+        ):
+            to_release.append(
+                self._query_nums_and_release_times[self._current_release_pointer]
+            )
+            self._current_release_pointer += 1
+
+        if (
+            self._current_release_pointer >= len(self._query_nums_and_release_times)
+            and len(to_release) == 0
+        ):
+            # Nothing left to release
+            return None
+
+        for i, (q, t) in enumerate(to_release):
+            query_name = f"Q{q}"
+            task_graph = self._task_graph_generators[query_name](
+                idx=i,
+                current_time=current_time,
+                start_time=t,
+            )
+            self._workload.add_task_graph(task_graph)
+
+        return self._workload
+
+
+def make_release_policy(
+    release_policy, release_policy_args, rng, seed, randomize_start_time=(0, 0)
+):
+    # Check that none of the arg values are None
+    assert all([val is not None for val in release_policy_args.values()])
+
+    # Construct the release policy
+    start_time = EventTime(
+        time=rng.randint(*randomize_start_time),
+        unit=EventTime.Unit.US,
+    )
+    release_policy = getattr(JobGraph.ReleasePolicy, release_policy)(
+        start=start_time, rng_seed=seed, **release_policy_args
+    )
+
+    return release_policy
+
+
+# TODO: make configurable
+TPCH_SUBDIR = "100g/"
+DECIMA_TPCH_DIR = (
+    "/home/dgarg39/erdos-scheduling-simulator/profiles/workload/tpch/decima/"
+)
+CLOUDLAB_TPCH_DIR = (
+    "/home/dgarg39/erdos-scheduling-simulator/profiles/workload/tpch/cloudlab/"
+)
+
+
+class SetWithCount(object):
+    """
+    allow duplication in set
+    """
+
+    def __init__(self):
+        self.set = {}
+
+    def __contains__(self, item):
+        return item in self.set
+
+    def add(self, item):
+        if item in self.set:
+            self.set[item] += 1
+        else:
+            self.set[item] = 1
+
+    def clear(self):
+        self.set.clear()
+
+    def remove(self, item):
+        self.set[item] -= 1
+        if self.set[item] == 0:
+            del self.set[item]
```

Comment:
Is this doing anything different from [`collections.Counter`](https://docs.python.org/3/library/collections.html#collections.Counter)?